tests/datasource/fluent/integration/test_sql_datasources.py

```diff
@@ -483,6 +482,64 @@ def test_sqlite(
 
         sqlite_ds.add_table_asset(asset_name, table_name=table_name)
 
+    @pytest.mark.parametrize(
+        "datasource_type,schema",
+        [
+            param("trino", None, marks=[pytest.mark.trino]),
+            param("postgres", None, marks=[pytest.mark.postgresql]),
+            param("snowflake", RAND_SCHEMA, marks=[pytest.mark.snowflake]),
+            param(
+                "databricks_sql",
+                RAND_SCHEMA,
+                marks=[pytest.mark.databricks],
+            ),
+            param("sqlite", None, marks=[pytest.mark.sqlite]),
+        ],
+    )
+    def test_checkpoint_run(
+        self,
+        request: pytest.FixtureRequest,
+        context: EphemeralDataContext,
+        table_factory: TableFactory,
+        asset_name: TableNameCase,
+        datasource_type: DatabaseType,
+        schema: str | None,
+    ):
+        datasource: SQLDatasource = request.getfixturevalue(f"{datasource_type}_ds")
+
+        table_name: str | None = TABLE_NAME_MAPPING[datasource_type].get(asset_name)
+        if not table_name:
+            pytest.skip(f"no '{asset_name}' table_name for {datasource_type}")
+
+        # create table
+        table_factory(
+            gx_engine=datasource.get_execution_engine(),
+            table_names={table_name},
+            schema=schema,
+        )
+
+        asset = datasource.add_table_asset(asset_name, table_name=table_name, schema_name=schema)
+        batch_definition = asset.add_batch_definition_whole_table("whole table!")
+
+        suite = context.add_expectation_suite(
+            expectation_suite_name=f"{datasource.name}-{asset.name}"
+        )
+        suite.add_expectation(gxe.ExpectColumnValuesToNotBeNull(column="name", mostly=1))
+        suite = context.add_or_update_expectation_suite(expectation_suite=suite)
+
+        checkpoint = context.checkpoints.add(
+            Checkpoint(
+                name=f"{datasource.name}-{asset.name}",
+                validation_definitions=[
+                    ValidationDefinition(name="foo", suite=suite, data=batch_definition)
```

Comment:
Nit - could we rename to something more specific to validation def? Just in case we have loggers or debuggers that could use that context?